File: bayan/bayan/foundation_vocabulary_complete.py
# -*- coding: utf-8 -*-
"""
إكمال قاموس الكلمات الأساسية - Complete Foundation Vocabulary

هذا الملف يحتوي على باقي الكلمات الأساسية

المصدر الأصلي: TypeScript في /vocabulary/foundationVocabularyComplete.ts
Original source: TypeScript in /vocabulary/foundationVocabularyComplete.ts
"""

from .foundation_vocabulary import (
    FoundationVocabulary,
    FoundationWord,
    FoundationWordType,
    FoundationCategory
)


def add_complete_vocabulary(vocab: FoundationVocabulary) -> None:
    """
    إضافة الكلمات الأساسية المتبقية
    Add remaining foundation words
    """
    
    # 15. المعرفة والشك - Knowledge & Doubt
    vocab.add_word(FoundationWord(
        arabic='شك',
        english='doubt',
        word_type=FoundationWordType.STATE,
        category=FoundationCategory.PSYCHOLOGICAL,
        core_meaning='عدم اليقين',
        related_words=['شكوك', 'مشكوك', 'ارتياب'],
        examples=['لديه شك', 'شك في الأمر'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='يقين',
        english='certainty',
        word_type=FoundationWordType.STATE,
        category=FoundationCategory.PSYCHOLOGICAL,
        core_meaning='العلم الثابت',
        related_words=['متيقن', 'تيقن', 'قطع'],
        examples=['لديه يقين', 'على يقين'],
        weight=0.9
    ))
    
    # 16. الأنشطة - Activities
    vocab.add_word(FoundationWord(
        arabic='ممارسة',
        english='practice',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='القيام بعمل بانتظام',
        related_words=['مارس', 'ممارس', 'مران'],
        examples=['ممارسة الرياضة', 'يمارس العمل'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='لعب',
        english='play',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='النشاط للمتعة',
        related_words=['لعبة', 'لاعب', 'ملعب'],
        examples=['لعب الكرة', 'لعبة ممتعة'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='مشي',
        english='walk',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='الحركة على القدمين',
        related_words=['مشى', 'ماشي', 'مشاء'],
        examples=['مشى في الطريق', 'المشي مفيد'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='ركض',
        english='run',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='الحركة السريعة على القدمين',
        related_words=['ركض', 'راكض', 'عدو'],
        root_word='ركض',
        examples=['ركض بسرعة', 'ركض في السباق'],
        weight=1.0
    ))
    
    # 17. أعضاء الجسم - Body Parts
    body_parts = [
        ('يد', 'hand', 'العضو للإمساك', ['أيدي', 'كف', 'أصابع'], ['رفع يده', 'يد قوية']),
        ('رجل', 'leg', 'العضو للمشي', ['أرجل', 'ساق', 'فخذ'], ['كسرت رجله', 'رجل طويلة']),
        ('قدم', 'foot', 'نهاية الرجل', ['أقدام', 'قدمان', 'كعب'], ['غسل قدميه', 'قدم كبيرة']),
        ('وجه', 'face', 'مقدمة الرأس', ['وجوه', 'محيا', 'سحنة'], ['وجه جميل', 'غسل وجهه']),
        ('عين', 'eye', 'العضو للرؤية', ['عيون', 'أعين', 'بصر'], ['عين زرقاء', 'فتح عينيه']),
        ('أنف', 'nose', 'العضو للشم', ['أنوف', 'منخر', 'خيشوم'], ['أنف طويل', 'شم بأنفه']),
        ('أذن', 'ear', 'العضو للسمع', ['آذان', 'أذنان', 'سمع'], ['أذن كبيرة', 'سمع بأذنيه']),
    ]
    
    for arabic, english, meaning, related, examples in body_parts:
        vocab.add_word(FoundationWord(
            arabic=arabic,
            english=english,
            word_type=FoundationWordType.ENTITY,
            category=FoundationCategory.PHYSICAL,
            core_meaning=meaning,
            related_words=related,
            examples=examples,
            weight=1.0
        ))
    
    # 18. البيئة الطبيعية - Natural Environment
    vocab.add_word(FoundationWord(
        arabic='مطر',
        english='rain',
        word_type=FoundationWordType.ENTITY,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='الماء النازل من السماء',
        related_words=['قطر', 'أمطار', 'ممطر'],
        examples=['نزل المطر', 'مطر غزير'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='قطر',
        english='drip',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='نزول الماء قطرة قطرة',
        related_words=['قطرة', 'تقطير', 'قاطر'],
        examples=['قطر الماء', 'قطرة ماء'],
        weight=0.8
    ))
    
    vocab.add_word(FoundationWord(
        arabic='زرع',
        english='plant',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='وضع البذور في الأرض',
        related_words=['زراعة', 'زارع', 'مزروع'],
        examples=['زرع الحقل', 'زراعة القمح'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='شجر',
        english='tree',
        word_type=FoundationWordType.ENTITY,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='النبات الكبير ذو الجذع',
        related_words=['شجرة', 'أشجار', 'شجيرة'],
        examples=['شجرة كبيرة', 'غابة أشجار'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='نبات',
        english='vegetation',
        word_type=FoundationWordType.ENTITY,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='ما ينبت من الأرض',
        related_words=['نبت', 'نباتات', 'نابت'],
        examples=['نبات أخضر', 'نباتات الحديقة'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='نما',
        english='grow',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='الزيادة في الحجم',
        related_words=['نمو', 'نامي', 'منمي'],
        examples=['نما الطفل', 'نمو سريع'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='حيا',
        english='live',
        word_type=FoundationWordType.STATE,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='الوجود بالحياة',
        related_words=['حياة', 'حي', 'أحياء'],
        examples=['حيا سنوات', 'حياة طويلة'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='مات',
        english='die',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='انتهاء الحياة',
        related_words=['موت', 'ميت', 'أموات'],
        examples=['مات الرجل', 'موت مفاجئ'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='ازدهر',
        english='flourish',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='النمو والتقدم',
        related_words=['ازدهار', 'مزدهر', 'زهر'],
        examples=['ازدهرت الحضارة', 'ازدهار اقتصادي'],
        weight=0.8
    ))
    
    vocab.add_word(FoundationWord(
        arabic='بحر',
        english='sea',
        word_type=FoundationWordType.ENTITY,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='الماء الكثير المالح',
        related_words=['بحار', 'بحري', 'محيط'],
        examples=['البحر واسع', 'سافر في البحر'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='نهر',
        english='river',
        word_type=FoundationWordType.ENTITY,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='الماء الجاري العذب',
        related_words=['أنهار', 'نهري', 'جدول'],
        examples=['نهر النيل', 'عبر النهر'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='شاطئ',
        english='shore',
        word_type=FoundationWordType.ENTITY,
        category=FoundationCategory.NATURAL_ENVIRONMENT,
        core_meaning='حافة البحر أو النهر',
        related_words=['شواطئ', 'ساحل', 'ضفة'],
        examples=['شاطئ البحر', 'جلس على الشاطئ'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='سبح',
        english='swim',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='الحركة في الماء',
        related_words=['سباحة', 'سابح', 'سبّاح'],
        examples=['سبح في البحر', 'سباحة ماهرة'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='غاص',
        english='dive',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='النزول تحت الماء',
        related_words=['غوص', 'غواص', 'غائص'],
        examples=['غاص في البحر', 'غوص عميق'],
        weight=0.8
    ))
    
    vocab.add_word(FoundationWord(
        arabic='غرق',
        english='drown',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='الموت في الماء',
        related_words=['غريق', 'إغراق', 'مغرق'],
        examples=['غرق في البحر', 'غريق البحر'],
        weight=0.8
    ))
    
    vocab.add_word(FoundationWord(
        arabic='نجا',
        english='survive',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='الخلاص من الخطر',
        related_words=['نجاة', 'ناجي', 'منجي'],
        examples=['نجا من الغرق', 'نجاة معجزة'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='استحم',
        english='bathe',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='غسل الجسم بالماء',
        related_words=['استحمام', 'حمام', 'محتم'],
        examples=['استحم بالماء', 'استحمام منعش'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='نشف',
        english='dry',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='إزالة الماء',
        related_words=['تنشيف', 'ناشف', 'جفاف'],
        examples=['نشف الملابس', 'تنشيف الشعر'],
        weight=0.8
    ))
    
    vocab.add_word(FoundationWord(
        arabic='جف',
        english='become dry',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='ذهاب الماء',
        related_words=['جفاف', 'جاف', 'يابس'],
        examples=['جف الماء', 'جفاف الأرض'],
        weight=0.9
    ))
    
    # 19. الحركة والانتقال - Movement & Transition
    vocab.add_word(FoundationWord(
        arabic='غاب',
        english='disappear',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.TRANSFORMATIONS,
        core_meaning='الاختفاء عن النظر',
        related_words=['غياب', 'غائب', 'مغيب'],
        examples=['غاب عن الأنظار', 'غياب طويل'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='حضر',
        english='attend',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='الوجود في المكان',
        related_words=['حضور', 'حاضر', 'محضر'],
        examples=['حضر الاجتماع', 'حضور قوي'],
        weight=0.9
    ))
    
    vocab.add_word(FoundationWord(
        arabic='ذهب',
        english='go',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='الانتقال من مكان',
        related_words=['ذهاب', 'ذاهب', 'مذهب'],
        examples=['ذهب إلى المدرسة', 'ذهاب وإياب'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='رجع',
        english='return',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='العودة إلى المكان',
        related_words=['رجوع', 'راجع', 'مرجع'],
        examples=['رجع إلى البيت', 'رجوع سريع'],
        weight=1.0
    ))
    
    vocab.add_word(FoundationWord(
        arabic='عاد',
        english='come back',
        word_type=FoundationWordType.ACTION,
        category=FoundationCategory.BASIC_ACTIONS,
        core_meaning='الرجوع مرة أخرى',
        related_words=['عودة', 'عائد', 'معاد'],
        examples=['عاد من السفر', 'عودة ميمونة'],
        weight=1.0
    ))

File: vocabulary/foundationVocabularyComplete.ts
/**
 * إكمال قاموس الكلمات الأساسية - Complete Foundation Vocabulary
 * 
 * هذا الملف يحتوي على باقي الكلمات الأساسية
 */

import { FoundationWord, FoundationWordType, FoundationCategory, FoundationVocabulary } from './foundationVocabulary';

/**
 * إضافة الكلمات الأساسية المتبقية
 * Add remaining foundation words
 */
export function addCompleteVocabulary(vocab: FoundationVocabulary): void {
  // 15. المعرفة والشك - Knowledge & Doubt
  vocab.addWord({
    arabic: 'شك',
    english: 'doubt',
    type: FoundationWordType.STATE,
    category: FoundationCategory.PSYCHOLOGICAL,
    coreMeaning: 'عدم اليقين',
    relatedWords: ['شكوك', 'مشكوك', 'ارتياب'],
    examples: ['لديه شك', 'شك في الأمر'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'يقين',
    english: 'certainty',
    type: FoundationWordType.STATE,
    category: FoundationCategory.PSYCHOLOGICAL,
    coreMeaning: 'العلم الثابت',
    relatedWords: ['متيقن', 'تيقن', 'قطع'],
    examples: ['لديه يقين', 'على يقين'],
    weight: 0.9
  });

  // 16. الأنشطة - Activities
  vocab.addWord({
    arabic: 'ممارسة',
    english: 'practice',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'القيام بعمل بانتظام',
    relatedWords: ['مارس', 'ممارس', 'مران'],
    examples: ['ممارسة الرياضة', 'يمارس العمل'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'لعب',
    english: 'play',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'النشاط للمتعة',
    relatedWords: ['لعبة', 'لاعب', 'ملعب'],
    examples: ['لعب الكرة', 'لعبة ممتعة'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'مشي',
    english: 'walk',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'الحركة على القدمين',
    relatedWords: ['مشى', 'ماشي', 'مشاء'],
    examples: ['مشى في الطريق', 'المشي مفيد'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'ركض',
    english: 'run',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'الحركة السريعة على القدمين',
    relatedWords: ['ركض', 'راكض', 'عدو'],
    examples: ['ركض بسرعة', 'ركض في السباق'],
    weight: 1.0
  });

  // 17. أعضاء الجسم - Body Parts
  vocab.addWord({
    arabic: 'يد',
    english: 'hand',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'العضو للإمساك',
    relatedWords: ['أيدي', 'كف', 'أصابع'],
    examples: ['رفع يده', 'يد قوية'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'رجل',
    english: 'leg',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'العضو للمشي',
    relatedWords: ['أرجل', 'ساق', 'فخذ'],
    examples: ['كسرت رجله', 'رجل طويلة'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'قدم',
    english: 'foot',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'نهاية الرجل',
    relatedWords: ['أقدام', 'قدمان', 'كعب'],
    examples: ['غسل قدميه', 'قدم كبيرة'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'وجه',
    english: 'face',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'مقدمة الرأس',
    relatedWords: ['وجوه', 'محيا', 'سحنة'],
    examples: ['وجه جميل', 'غسل وجهه'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'عين',
    english: 'eye',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'العضو للرؤية',
    relatedWords: ['عيون', 'أعين', 'بصر'],
    examples: ['عين زرقاء', 'فتح عينيه'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'أنف',
    english: 'nose',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'العضو للشم',
    relatedWords: ['أنوف', 'منخر', 'خيشوم'],
    examples: ['أنف طويل', 'شم بأنفه'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'أذن',
    english: 'ear',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.PHYSICAL,
    coreMeaning: 'العضو للسمع',
    relatedWords: ['آذان', 'أذنان', 'سمع'],
    examples: ['أذن كبيرة', 'سمع بأذنيه'],
    weight: 1.0
  });

  // 18. البيئة الطبيعية - Natural Environment
  vocab.addWord({
    arabic: 'مطر',
    english: 'rain',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'الماء النازل من السماء',
    relatedWords: ['قطر', 'أمطار', 'ممطر'],
    examples: ['نزل المطر', 'مطر غزير'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'قطر',
    english: 'drip',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'نزول الماء قطرة قطرة',
    relatedWords: ['قطرة', 'تقطير', 'قاطر'],
    examples: ['قطر الماء', 'قطرة ماء'],
    weight: 0.8
  });

  vocab.addWord({
    arabic: 'زرع',
    english: 'plant',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'وضع البذور في الأرض',
    relatedWords: ['زراعة', 'زارع', 'مزروع'],
    examples: ['زرع الحقل', 'زراعة القمح'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'شجر',
    english: 'tree',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'النبات الكبير ذو الجذع',
    relatedWords: ['شجرة', 'أشجار', 'شجيرة'],
    examples: ['شجرة كبيرة', 'غابة أشجار'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'نبات',
    english: 'vegetation',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'ما ينبت من الأرض',
    relatedWords: ['نبت', 'نباتات', 'نابت'],
    examples: ['نبات أخضر', 'نباتات الحديقة'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'نما',
    english: 'grow',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'الزيادة في الحجم',
    relatedWords: ['نمو', 'نامي', 'منمي'],
    examples: ['نما الطفل', 'نمو سريع'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'حيا',
    english: 'live',
    type: FoundationWordType.STATE,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'الوجود بالحياة',
    relatedWords: ['حياة', 'حي', 'أحياء'],
    examples: ['حيا سنوات', 'حياة طويلة'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'مات',
    english: 'die',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'انتهاء الحياة',
    relatedWords: ['موت', 'ميت', 'أموات'],
    examples: ['مات الرجل', 'موت مفاجئ'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'ازدهر',
    english: 'flourish',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'النمو والتقدم',
    relatedWords: ['ازدهار', 'مزدهر', 'زهر'],
    examples: ['ازدهرت الحضارة', 'ازدهار اقتصادي'],
    weight: 0.8
  });

  vocab.addWord({
    arabic: 'بحر',
    english: 'sea',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'الماء الكثير المالح',
    relatedWords: ['بحار', 'بحري', 'محيط'],
    examples: ['البحر واسع', 'سافر في البحر'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'نهر',
    english: 'river',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'الماء الجاري العذب',
    relatedWords: ['أنهار', 'نهري', 'جدول'],
    examples: ['نهر النيل', 'عبر النهر'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'شاطئ',
    english: 'shore',
    type: FoundationWordType.ENTITY,
    category: FoundationCategory.NATURAL_ENVIRONMENT,
    coreMeaning: 'حافة البحر أو النهر',
    relatedWords: ['شواطئ', 'ساحل', 'ضفة'],
    examples: ['شاطئ البحر', 'جلس على الشاطئ'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'سبح',
    english: 'swim',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'الحركة في الماء',
    relatedWords: ['سباحة', 'سابح', 'سبّاح'],
    examples: ['سبح في البحر', 'سباحة ماهرة'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'غاص',
    english: 'dive',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'النزول تحت الماء',
    relatedWords: ['غوص', 'غواص', 'غائص'],
    examples: ['غاص في البحر', 'غوص عميق'],
    weight: 0.8
  });

  vocab.addWord({
    arabic: 'غرق',
    english: 'drown',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'الموت في الماء',
    relatedWords: ['غريق', 'إغراق', 'مغرق'],
    examples: ['غرق في البحر', 'غريق البحر'],
    weight: 0.8
  });

  vocab.addWord({
    arabic: 'نجا',
    english: 'survive',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'الخلاص من الخطر',
    relatedWords: ['نجاة', 'ناجي', 'منجي'],
    examples: ['نجا من الغرق', 'نجاة معجزة'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'استحم',
    english: 'bathe',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'غسل الجسم بالماء',
    relatedWords: ['استحمام', 'حمام', 'محتم'],
    examples: ['استحم بالماء', 'استحمام منعش'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'نشف',
    english: 'dry',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'إزالة الماء',
    relatedWords: ['تنشيف', 'ناشف', 'جفاف'],
    examples: ['نشف الملابس', 'تنشيف الشعر'],
    weight: 0.8
  });

  vocab.addWord({
    arabic: 'جف',
    english: 'become dry',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'ذهاب الماء',
    relatedWords: ['جفاف', 'جاف', 'يابس'],
    examples: ['جف الماء', 'جفاف الأرض'],
    weight: 0.9
  });

  // 19. الحركة والانتقال - Movement & Transition
  vocab.addWord({
    arabic: 'غاب',
    english: 'disappear',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.TRANSFORMATIONS,
    coreMeaning: 'الاختفاء عن النظر',
    relatedWords: ['غياب', 'غائب', 'مغيب'],
    examples: ['غاب عن الأنظار', 'غياب طويل'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'حضر',
    english: 'attend',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'الوجود في المكان',
    relatedWords: ['حضور', 'حاضر', 'محضر'],
    examples: ['حضر الاجتماع', 'حضور قوي'],
    weight: 0.9
  });

  vocab.addWord({
    arabic: 'ذهب',
    english: 'go',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'الانتقال من مكان',
    relatedWords: ['ذهاب', 'ذاهب', 'مذهب'],
    examples: ['ذهب إلى المدرسة', 'ذهاب وإياب'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'رجع',
    english: 'return',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'العودة إلى المكان',
    relatedWords: ['رجوع', 'راجع', 'مرجع'],
    examples: ['رجع إلى البيت', 'رجوع سريع'],
    weight: 1.0
  });

  vocab.addWord({
    arabic: 'عاد',
    english: 'come back',
    type: FoundationWordType.ACTION,
    category: FoundationCategory.BASIC_ACTIONS,
    coreMeaning: 'الرجوع مرة أخرى',
    relatedWords: ['عودة', 'عائد', 'معاد'],
    examples: ['عاد من السفر', 'عودة ميمونة'],
    weight: 1.0
  });
}

